cpp/test/Ice/idleTimeout/TestI.cpp

```diff
@@ -19,3 +21,21 @@ TestIntfI::shutdown(const Ice::Current& current)
 {
     current.adapter->getCommunicator()->shutdown();
 }
+
+void
+TestIntfI::setCallback(std::optional<Test::TestIntfPrx> prx, const Ice::Current& current)
+{
+    // Call from a detach thread to avoid blocking the server thread
```

Comment:
It would be simpler to make a "blocking" oneway call.

Reply:
Why do you want to check the exception? In your current code, you ignore it.

Reply:
There is a missed `test(false);` to ensure the connection is closed by the peer idle check.

Reply:
In this case, I would not start a thread but call sleepAsync "with lambdas". You can put the test in the response/exception callbacks.